pkg/operators/clustermanager/controllers/clustermanagercontroller/clustermanager_controller.go

```diff
@@ -487,17 +528,34 @@ func applyHubResources(
 			helpers.SetRelatedResourcesStatusesWithObj(relatedResources, objData)
 			return objData, nil
 		},
-		hubApiserviceFiles...,
+		webhookResourceFiles...,
 	)
-	for _, result := range apiserviceResults {
+	for _, result := range resourceResults {
 		if result.Error != nil {
 			appliedErrs = append(appliedErrs, fmt.Errorf("%q (%T): %v", result.File, result.Type, result.Error))
 		}
 	}
-
 	return appliedErrs, nil
 }
 
+func checkWebhookPodRunning(managementKubeClient kubernetes.Interface, manifestsConfig manifests.HubConfig, webhookName string) error {
+	webhookDeployment, err := managementKubeClient.AppsV1().Deployments(manifestsConfig.ClusterManagerNamespace).Get(context.Background(),
+		webhookName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	if webhookDeployment.Generation != webhookDeployment.Status.ObservedGeneration {
+		return fmt.Errorf("New %v webhook deployment not observed. Expect generation:%v, Observed Generation:%v", webhookName, webhookDeployment.Generation, webhookDeployment.Status.ObservedGeneration)
```

Comment:
shouldn't we requeue with fallback instead of returning error?

Reply:
done